packages/connection-form/src/components/connect-form.tsx

```diff
@@ -296,6 +310,10 @@ function ConnectForm({
                 ...favoriteInfo,
               },
             });
+
+            if (saveConnectionModal === 'saveAndConnect') {
+              onSubmitForm();
```

Comment:
I think the connection options in the form don't have the recently set favorite settings on the first connection.
After save and connect with a new connection:
(Lack of connection name)
<img width="1015" alt="Screen Shot 2022-06-14 at 1 45 02 PM" src="https://user-images.githubusercontent.com/1791149/173656882-422e0688-0459-45d4-a10c-85d732f15c8d.png">
After reloading the favorites list and connecting to the favorite:
(has connection name)
<img width="891" alt="Screen Shot 2022-06-14 at 1 45 07 PM" src="https://user-images.githubusercontent.com/1791149/173656917-481726f9-2956-4813-b78d-2f652640cf7c.png">

Not sure this is a blocker, but I could see it tripping up some folks. 

Due to the nature of how the `useConnectForm` is setup it may take some extra work to have it update the current connection options and then connect from there. Maybe we could add an optional argument overload to the `onSubmitForm` function to explicitly connect to the passed options? Totally up to you implementation wise, just suggesting a thought.

Reply:
Same as this? https://mongodb.slack.com/archives/GDTJXPHD0/p1654259449438809

Reply:
I think this is an existing problem. ie if you hit save via the favorite modal and then connect on main now it also happens.

Due to that I'm not sure what's best practise - make this PR larger and fix it now or fix it in a followup?

Reply:
Looks like it's the same, yes. When saving happens we don't update the current connection settings. As for best practice, it's up to you, since it won't change this code much I think it would be nice to have it in another pr. It could be a separate `fix` then.

Reply:
https://github.com/mongodb-js/compass/pull/3189